### PR TITLE
foot note adjustments

### PR DESCRIPTION
**Pull request by facelessuser**
Add a config to set the footnote separator. Also remove rel/rev as they aren't really compatible with HTML5 and we already have classes set for refs and backrefs. Ref #723.

### CODE REVIEW

File: tests/test_extensions.py [hunk 1]
```diff
@@ -1121,14 +1121,35 @@ def testBacklinkText(self):
...
             '<li id="fn:1">\n'
-            '<p>A Footnote&#160;<a class="footnote-backref" href="#fnref:1" rev="footnote"'
+            '<p>A Footnote&#160;<a class="footnote-backref" href="#fnref:1"'
             ' title="Jump back to footnote 1 in the text">back</a></p>\n'
             '</li>\n'
             '</ol>\n'
             '</div>'
         )
+
+    def testFootnoteSeparator(self):
```
Comment (waylan) [inline tests/test_extensions.py]: Let's add this new test using the new testing framework. Ideally the existing tests should be moved as well. But I'll settle for the new one only for now.

Comment (facelessuser) [reply tests/test_extensions.py]: I'll most likely move all footnote tests in this file to the new format.

**Comment by facelessuser** [comment]
Requested changes have been made.

**Comment by minchinweb** [comment]
Out of curiosity, is there an ETA for when a release containing this fix will be pushed out?

**Comment by facelessuser** [comment]
I am hoping soon, not specifically for this issue, but for the code escape issue: https://github.com/Python-Markdown/markdown/issues/725.

**Comment by waylan** [comment]
I expect a release once #732 is finalized (which includes some maintenance to bring the repo inline with the guidelines).

Note that this adds a feature, so this cannot be part of a bugfix release, but will be a point release (3.1). Therefore, it is a little slower moving.

**Comment by justinmayer** [comment]
@waylan: Any chance that a v3.0.2 release could be published sooner that contains the aforementioned fixes? (i.e., without new features)

**Comment by waylan** [comment]
@justinmayer that is a valid question. However, the effort to backport the bugfixes and the effort to finalize 3.1 are probably about the same. Therefore, I think it makers more sense to move forward with 3.1. It's just the matter of finding the time.

**Pull request merged**